keras/applications/imagenet_utils.py

```diff
@@ -58,6 +49,75 @@ def preprocess_input(x, data_format=None, mode='caffe'):
     return x
 
 
+def _bias_add_match_type(x, bias):
+    if K.dtype(x) != K.dtype(bias):
+        x = K.bias_add(x, K.cast(bias, K.dtype(x)))
+    else:
+        x = K.bias_add(x, bias)
+    return x
+
+
+def _preprocess_symbolic_input(x, data_format, mode):
+    global _IMAGENET_MEAN_CHW
+    global _IMAGENET_MEAN_HWC
+
+    if data_format is None:
+        data_format = K.image_data_format()
+    assert data_format in {'channels_last', 'channels_first'}
+
+    if mode == 'tf':
+        x /= 255.
+        x -= 0.5
+        x *= 2.
+        return x
+
+    imagenet_mean = [103.939, 116.779, 123.68]
+    if data_format == 'channels_first':
+        # 'RGB'->'BGR'
+        if K.ndim(x) == 3:
+            x = x[::-1, ...]
+        else:
+            x = x[:, ::-1, ...]
+        # Zero-center by mean pixel
+        if _IMAGENET_MEAN_CHW is None:
+            _IMAGENET_MEAN_CHW = K.constant(
+                -np.array(imagenet_mean).reshape((3, 1, 1)))
```

Comment:
Since you're using `bias_add`, you don't need to reshape. It does it already. `(3,)` will work. For the same reason, you no longer need to have 2 global variables, one will suffice.